plugins/org.python.pydev.core/src/org/python/pydev/core/docutils/ReversePyDocIterator.java

```diff
@@ -0,0 +1,232 @@
+            } catch (BadLocationException e) {
+                //ignore (keep the last one)
+            }
+        }
+        return lastReturned;
+    }
+
+    private String nextInLiteral() {
+        StringBuffer buf = new StringBuffer();
+        try {
+
+            char ch = doc.getChar(offset);
+            if (ch == '\n') {
+                offset--;
+                if (doc.getChar(offset) == '\r') {
+                    offset--;
+                }
+                buf.append(ch);
+                ch = doc.getChar(offset);
+            }
+
+            while (offset >= literalStart && ch != '\n' && ch != '\r') {
+                ch = doc.getChar(offset);
+                if (ch != '\n' && ch != '\r') {
+                    offset--;
+                    if (changeLiteralsForSpaces) {
+                        buf.insert(0, ' ');
+                    }
+                }
+            }
+            if (offset < literalStart) {
+                inLiteral = false;
+            }
+            return buf.toString();
+
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * @return the next line in the document
+     */
+    @Override
+    public String next() {
+
+        try {
+            StringBuffer buf = new StringBuffer();
+
+            if (inLiteral) {
+                int initialOffset = offset;
+                String ret = nextInLiteral();
+                if (ret.length() > 0 && initialOffset >= offset) { //if it didn't move in the offset, disregard the results
+                    if (StringUtils.endsWith(ret, '\r') || StringUtils.endsWith(ret, '\n')) {
+                        if (!addNewLinesToRet) {
+                            ret = ret.substring(0, ret.length() - 1);
+                        }
+                        buf.insert(0, ret);
+                        if (inLiteral) {
+                            return ret;
+                        }
+                    } else {
+                        buf.insert(0, ret);
+                    }
+                }
+            }
+
+            char ch = doc.getChar(offset);
+
+            //handle the \r, \n or \r\n
+            if (ch == '\n' || ch == '\r') {
+                if (addNewLinesToRet) {
+                    buf.append(ch);
+                }
+                offset--;
+                ch = doc.getChar(offset);
+                if (ch == '\n') {
+                    if (offset >= 0 && ch == '\r') {
+                        offset--;
+                        ch = doc.getChar(offset);
+                        if (addNewLinesToRet) {
+                            buf.append('\n');
+                        }
+                    }
+                }
+
+            }
+
+            ParsingUtils parsingUtils = ParsingUtils.create(doc);
+            while (ch != '\r' && ch != '\n' && offset >= 0) {
+                ch = doc.getChar(offset);
+                if (ch == '#') {
```

Comment:
This doesn't seem right... please create a test to check comments (dealing with comments is a bit more tricky -- for literals you can go to the literal start point, but for the comment you actually need to get the whole line first and check for the `#` and go to the end of the line).

As a note, make sure you test a literal with a comment inside too.

Something as:

```
#this should be ignored ''''
s = '''
# inside string (not a comment) '''
'''
```